app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -70,12 +94,146 @@ public void onCreate(Bundle savedInstanceState) {
             LocationTracker.getInstance().start(this);
         }
 
+        handleRecording(getIntent());
+    }
+
+    private void handleRecording(Intent intent) {
+        if (intent == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent instance");
+            finishActivity();
+            return;
+        }
+
+        String recordingAction = intent.getAction();
+        if (recordingAction == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent.action instance");
+            finishActivity();
+            return;
+        }
+
+        if (!recordingAction.startsWith(ACTION_RECORDING_BASE)) {
+            Log.i(TAG, "handleRecording: Received different intent with action: "
+                    + recordingAction);
+            finishActivity();
+            return;
+        }
+
+        if (RecorderUtil.isLowerThanQ()) {
+            Log.e(TAG, "handleRecording: Current Android OS Version is lower than Q");
+            finishActivity();
+            return;
+        }
+
+        if (!RecorderUtil.areRecordingPermissionsGranted(getApplicationContext())) {
+            Log.e(TAG, "handleRecording: Required permissions are not granted");
+            finishActivity();
+            return;
+        }
+
+        if (recordingAction.equals(ACTION_RECORDING_START)) {
+            String recordingFilename = intent.getStringExtra(ACTION_RECORDING_FILENAME);
```

Comment:
can it be null?

Reply:
yeah it can be but we are checking against it on here https://github.com/sirmordred/io.appium.settings/blob/master/app/src/main/java/io/appium/settings/recorder/RecorderUtil.java#L57 👍